src/core/execute.c

```diff
@@ -201,6 +202,63 @@ static const char *exec_context_tty_path(const ExecContext *context) {
         return "/dev/console";
 }
 
+static int exec_context_tty_size(const ExecContext *context, unsigned *ret_rows, unsigned *ret_cols) {
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        const char *tty;
+        int r;
+
+        assert(context);
+        assert(ret_rows);
+        assert(ret_cols);
+
+        tty = skip_dev_prefix(exec_context_tty_path(context));
+
+        if (context->tty_rows == UINT_MAX) {
+                _cleanup_free_ char *key = NULL, *value = NULL;
+
+                key = strjoin("systemd.tty.", tty, ".rows");
+                if (!key)
+                        return -ENOMEM;
+
+                r = proc_cmdline_get_key(key, 0, &value);
```

Comment:
This may call `proc_cmdline_get_key()` twice. How about to use `proc_cmdline_get_key_many()`?